civis/io/_files.py

```diff
@@ -148,7 +147,7 @@ def _gen_chunks(part_buf, max_bytes, chunk_size=32 * 1024):
     def _upload_part(item):
         part_num, part_url = item[0], item[1]
         log.debug('Uploading file part %s', part_num)
-        file_out = os.path.join(tmp_dir, str(part_num) + '.csv')
+        file_out = tmp_path.format(i)
```

Comment:
Does `tmp_path` need to be passed to this function? I'm not clear where it is defined.

Reply:
the _upload_part func is inline so its within the same scope

Reply:
So `tmp_path` is defined on L164, *after* this function is defined? I have feeling this might be confusing later.  Could we do this with a closure instead?  Something like

```python
def _create_upload_part(tmp_dir):
    def _upload_part(item):
        ...
    return _upload_part
```

Reply:
Or you could make `tmp_path` an input to this function and use functools.partial to include it.

Reply:
updated; implemented use of functools.partial